# PyCity Schools Analysis

In [5]:
# Dependencies and Setup
import pandas as pd
from pathlib import Path

# File to Load (Remember to Change These)
school_data_to_load = Path("../Resources/schools_complete.csv")
student_data_to_load = Path("../Resources/students_complete.csv")

# Read School and Student Data File and store into Pandas DataFrames
schools_df = pd.read_csv(school_data_to_load)
student_df = pd.read_csv(student_data_to_load)

# Combine the data into a single dataset.  
data_complete = pd.merge(student_df, schools_df, how="outer", on=["school_name", "school_name"])
schools_df



School ID            school_name      type  size   budget
0           0      Huang High School  District  2917  1910635
1           1   Figueroa High School  District  2949  1884411
2           2    Shelton High School   Charter  1761  1056600
3           3  Hernandez High School  District  4635  3022020
4           4    Griffin High School   Charter  1468   917500
5           5     Wilson High School   Charter  2283  1319574
6           6    Cabrera High School   Charter  1858  1081356
7           7     Bailey High School  District  4976  3124928
8           8     Holden High School   Charter   427   248087
9           9       Pena High School   Charter   962   585858
10         10     Wright High School   Charter  1800  1049400
11         11  Rodriguez High School  District  3999  2547363
12         12    Johnson High School  District  4761  3094650
13         13       Ford High School  District  2739  1763916
14         14     Thomas High School   Charter  1635  1043130

Huang, Figueroa, Shelton, Hernandez, Griffin, Wilson, Cabrera, Bailey, Holden, Pena, Wright, Rodriguez, Johnson, Ford, Thomas

There are 15 schools - store in a variable

In [6]:
school_count = len(schools_df["school_name"])
school_count

15

In [10]:
data_complete["Student ID"].value_counts()
data_complete.describe()



Student ID  reading_score    math_score     School ID          size  \
count  39170.000000    39170.00000  39170.000000  39170.000000  39170.000000   
mean   19584.500000       81.87784     78.985371      6.978172   3332.957110   
std    11307.549359       10.23958     12.309968      4.444329   1323.914069   
min        0.000000       63.00000     55.000000      0.000000    427.000000   
25%     9792.250000       73.00000     69.000000      3.000000   1858.000000   
50%    19584.500000       82.00000     79.000000      7.000000   2949.000000   
75%    29376.750000       91.00000     89.000000     11.000000   4635.000000   
max    39169.000000       99.00000     99.000000     14.000000   4976.000000   

             budget  
count  3.917000e+04  
mean   2.117241e+06  
std    8.749987e+05  
min    2.480870e+05  
25%    1.081356e+06  
50%    1.910635e+06  
75%    3.022020e+06  
max    3.124928e+06

length of the Student ID column is 39170   same for all columns

In [12]:
student_count = len(student_df["Student ID"].unique())
print(student_count)

student_name_count = len(student_df["student_name"].unique())
print(student_name_count)

39170
32715


there are fewer unique student names than there are student IDs
I'm sure that some students have the same name, but this is a big difference

In [15]:
data_inner = pd.merge(student_df, schools_df, on=["school_name"])
data_inner.head(10)


Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   
5           5      Bryan Miranda      M   9th  Huang High School   
6           6      Sheena Carter      F  11th  Huang High School   
7           7       Nicole Baker      F  12th  Huang High School   
8           8       Michael Roth      M  10th  Huang High School   
9           9     Matthew Greene      M  10th  Huang High School   

   reading_score  math_score  School ID      type  size   budget  
0             66          79          0  District  2917  1910635  
1             94          61          0  District  2917  1910635  
2             90          60          0  District  2917  1910635  
3             67          58          0  District  2917  1910635  
4             97          84          0  District  2917  1910635  
5             94          94          0  District  2917  1910635  
6             82          80          0  District  2917  1910635  
7             96          69          0  District  2917  1910635  
8             95          87          0  District  2917  1910635  
9             96          84          0  District  2917  1910635

get enrollment count at each school from the size column

In [20]:
enrollments= schools_df["size"].unique()
print(enrollments)
enrollments_total = enrollments.sum()
print(enrollments_total)


[2917 2949 1761 4635 1468 2283 1858 4976  427  962 1800 3999 4761 2739
 1635]
39170


the sum of the individual sizes of the schools is 39170
student_count = 39170

let's do the same thing for the budget

In [21]:
budgets_all = schools_df["budget"].unique()
print(budgets_all)

budgets_total = budgets_all.sum()
print(budgets_total)


[1910635 1884411 1056600 3022020  917500 1319574 1081356 3124928  248087
  585858 1049400 2547363 3094650 1763916 1043130]
24649428


There are 15 unique budget values.  This is as expected
The total is 24649428 stored in budgets_total

for the average math scores, we'll want to look at the merged datafile

In [23]:
avg_reading_score = data_complete["reading_score"].mean()
print(avg_reading_score)

avg_math_score = data_complete["math_score"].mean()
print(avg_math_score)

81.87784018381414
78.98537145774827


overall average math score stored in avg_math_score
overall average reading score stored in ave_reading_score

In [28]:

data_complete.dtypes

Student ID        int64
student_name     object
gender           object
grade            object
school_name      object
reading_score     int64
math_score        int64
School ID         int64
type             object
size              int64
budget            int64
dtype: object

Creating a new column for pass or fail math score with 60 and higher being pass and lower being fail
and one for reading also

In [36]:

data_complete["math_pass"] = data_complete["math_score"].apply(lambda x:"pass" if x>=60 else "fail")

data_complete["reading_pass"] = data_complete["reading_score"].apply(lambda x:"pass" if x>=60 else "fail")

math_num_pass = data_complete["math_pass"].value_counts()
reading_num_pass = data_complete["reading_pass"].value_counts()

print(math_num_pass)
print(reading_num_pass)

data_complete.head()

pass    36211
fail     2959
Name: math_pass, dtype: int64
pass    39170
Name: reading_pass, dtype: int64


Student ID       student_name gender grade        school_name  \
0           0       Paul Bradley      M   9th  Huang High School   
1           1       Victor Smith      M  12th  Huang High School   
2           2    Kevin Rodriguez      M  12th  Huang High School   
3           3  Dr. Richard Scott      M  12th  Huang High School   
4           4         Bonnie Ray      F   9th  Huang High School   

   reading_score  math_score  School ID      type  size   budget math_pass  \
0             66          79          0  District  2917  1910635      pass   
1             94          61          0  District  2917  1910635      pass   
2             90          60          0  District  2917  1910635      pass   
3             67          58          0  District  2917  1910635      fail   
4             97          84          0  District  2917  1910635      pass   

  reading_pass  
0         pass  
1         pass  
2         pass  
3         pass  
4         pass

get percents passing
math_percent_pass
reading_percent_pass
overall_percent_pass

In [39]:
math_percent_pass = (math_num_pass / student_count)*100
reading_percent_pass = (reading_num_pass / student_count)*100

def get_status(row):
    if row["math_pass"] == "pass" and row["reading_pass"] == "pass":
        return "pass"
    else:
        return "fail"

data_complete["overall pass"] = data_complete.apply(lambda row: get_status(row), axis=1)

overall_num_pass = data_complete["overall pass"].value_counts()

print(overall_num_pass)

overall_percent_pass = (overall_num_pass / student_count)*100

print(math_percent_pass)
print(reading_percent_pass)
print(overall_percent_pass)


pass    36211
fail     2959
Name: overall pass, dtype: int64
pass    92.445749
fail     7.554251
Name: math_pass, dtype: float64
pass    100.0
Name: reading_pass, dtype: float64
pass    92.445749
fail     7.554251
Name: overall pass, dtype: float64
